syntax/decode.go

```diff
@@ -138,37 +138,50 @@ func uintDecoder(d *decodeState, v reflect.Value, opts decOpts) int {
 }
 
 func varintDecoder(d *decodeState, v reflect.Value, opts decOpts) int {
+	l, val := readVarint(d)
+
+	uintLen := int(v.Elem().Type().Size())
+	if uintLen < l {
```

Comment:
Should this test for `==` instead of `<`?  Might be more conservative to pass the desired length into `readVarint` and have it fail if the varint isn't the right size.

Reply:
Comment withdrawn.  On further reflection, this is wrong.